wikidatarefisland/data_access/storage.py

```diff
@@ -15,11 +17,29 @@ def getLines(self, file_):
         path = os.path.join(self.path, file_)
         with open(path, 'r') as f:
             for line in f:
+                yield json.loads(line.strip())
+
+    def get_dump_lines(self, path):
+        mode = 'r'
+        file_ = os.path.split(path)[-1]
+        if file_.endswith('.gz'):
+            f = gzip.open(path, mode)
+        elif file_.endswith('.bz2'):
+            f = bz2.BZ2File(path, mode)
+        elif file_.endswith('.json'):
+            f = open(path, mode)
+        else:
+            raise NotImplementedError(f'Reading file {file_} is not supported')
+        try:
+            for line in f:
+                if isinstance(line, bytes):
+                    line = line.decode('utf-8')
                 try:
-                    # TODO Fix this hack to make json behave a bit like jsonl: T251271
                     yield json.loads(line.strip().strip(','))
```

Comment:
".strip(',')" doesn't remove anything if ',' doesn't exist in the end of the line. So it doesn't do anything if the line doesn't end with `,`

Reply:
I think the question is should this work with files that are formatted in a way that isn't exactly 1 item per line?

I could be convinced this is ok but it might be clearer if this function then only took a .gz file. Right now it looks a bit like in a dev set up you could dump a json array of items and it would work (but of course it would potentially try and feed non-item shaped objects into the pipe).

Other options I saw for doing this more robustly are:
`ijson` (which I dug up in the ticked but we thought much be too unused; I had a dig and actually it seemed like some "fairly" big projects depend on it)
`yajl` and it's python wrapper which seems to be the old but classic solution to streaming json using a small C program
or we could even shell to `jq` which has a `--stream` option

Reply:
Even the API returns everything in one line: https://www.wikidata.org/w/api.php?action=wbgetentities&ids=Q42&format=json
I understand the usefulness but I haven't found a real usecase for this. I'm not saying it's not valid but I think this is out of scope of this ticket and patch.

Reply:
I think we should probably get this merged but I don't think this does work for the api? e.g. https://www.wikidata.org/w/api.php?action=wbgetentities&ids=Q45374353|Q45374354&format=json looks to me like it is not split by line? (but also the entities are not provided as an array but rather as keys to the object)

I'll try and add a follow-up making this clear :)